api/dataprocessor.go

```diff
@@ -551,6 +551,7 @@ func (s *Server) getSeriesCachedStore(ctx *requestContext, until uint32) ([]chun
 		if cacheRes.From != cacheRes.Until {
 			storeIterGens, err := s.BackendStore.Search(ctx.ctx, ctx.AMKey, ctx.Req.TTL, cacheRes.From, cacheRes.Until)
 			if err != nil {
+				log.Warn("Failed to retrieve data for key = %s", ctx.AMKey.String())
```

Comment:
i'm generally against logging the same error twice.
note that this error bubbles up via getSeries, getSeriesFixed, getTarget, getTargetsLocal
and eventually gets logged in the two call sites (clustered and local): getData and executeplan. is this not sufficient?

Reply:
I'm not logging the error here, I'm logging the key. The error didn't include the key and I was trying to trace down the source.

Reply:
right.
i'm all for:
1) bypassing cache if we detect corruption
2) logging extra diagnostics so we can troubleshoot corruption cases

but does this logline really help with either of these? was it more of a temporary development thing that maybe doesn't need to go into this PR? if it does help for this use case, would it make sense to log the error also (even though it'll be logged in the caller as mentioned)

Reply:
I can remove it as it's not related to this PR. But as mentioned above it logs the key *not* the error.